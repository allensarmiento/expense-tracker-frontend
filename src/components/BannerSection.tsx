import { ReactNode } from 'react';
import styled from 'styled-components';

type SectionProps = {
  justifyContent?: string;
};

const Section = styled.section<SectionProps>`
  display: flex;
  flex-direction: column;
  justify-content: ${props => props.justifyContent || 'default'}
`;

type BannerSectionProps = {
  justifyContent?: string;
  children?: ReactNode;
};

const BannerSection = ({ justifyContent, children }: BannerSectionProps) => {
  return <Section justifyContent={justifyContent}>{children}</Section>
};

export default BannerSection;
