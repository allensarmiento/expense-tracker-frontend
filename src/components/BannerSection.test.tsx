import { shallow } from 'enzyme';
import BannerSection from './BannerSection';

it('renders without crashing', () => {
  expect(shallow(<BannerSection />)).toMatchSnapshot();

  expect(shallow(<BannerSection justifyContent="space-between" />))
    .toMatchSnapshot();

  expect(shallow(<BannerSection>Test</BannerSection>))
    .toMatchSnapshot();
});
