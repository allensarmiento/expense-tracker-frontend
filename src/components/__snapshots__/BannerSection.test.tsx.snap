// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `<styled.section />`;

exports[`renders without crashing 2`] = `
<styled.section
  justifyContent="space-between"
/>
`;

exports[`renders without crashing 3`] = `
<styled.section>
  Test
</styled.section>
`;
